programs/fileio.c

```diff
@@ -564,7 +564,7 @@ static unsigned long long FIO_compressGzFrame(cRess_t* ress,
                 strm.avail_out = (uInt)ress->dstBufferSize;
             }
         }
-        if (!srcFileSize)
+        if (srcFileSize != UTIL_FILESIZE_UNKNOWN)
```

Comment:
`srcFileSize == UTIL_FILESIZE_UNKNOWN` here and below.

Reply:
Nice catch !